hadoop-plugin/src/main/groovy/com/linkedin/gradle/hadoopdsl/checker/RequiredFieldsChecker.groovy

```diff
@@ -183,4 +186,76 @@ class RequiredFieldsChecker extends BaseStaticChecker {
     }
   }
+
+  @Override
+  void visitJob(HdfsToTeradataJob job) {
+    foundError |= validateNotEmpty(job, "hostName", job.hostName);
+    foundError |= validateNotEmpty(job, "userId", job.userId);
+    foundError |= validateNotEmpty(job, "credentialName", job.credentialName);
+    foundError |= validateNotEmpty(job, "sourceHdfsPath", job.sourceHdfsPath);
+    foundError |= validateNotEmpty(job, "targetTable", job.targetTable);
+
+    boolean isAvroSchemaExist = false;
+    isAvroSchemaExist ^= job.avroSchemaPath == null || job.avroSchemaPath.isEmpty();
+    isAvroSchemaExist ^= job.avroSchemaInline == null || job.avroSchemaInline.isEmpty();
+
+    if(!isAvroSchemaExist) {
+      project.logger.lifecycle("RequiredFieldsChecker ERROR: ${job.getClass().getSimpleName()} ${job.name} needs either avroSchemaPath or avroSchemaInline defined");
+      foundError = true;
+    }
+  }
+
+  @Override
+  void visitJob(TeradataToHdfsJob job) {
+    foundError |= validateNotEmpty(job, "hostName", job.hostName);
+    foundError |= validateNotEmpty(job, "userId", job.userId);
+    foundError |= validateNotEmpty(job, "credentialName", job.credentialName);
+    foundError |= validateNotEmpty(job, "targetHdfsPath", job.targetHdfsPath);
+
+    boolean isSourceExist = false;
+    isSourceExist ^= job.sourceTable == null || job.sourceTable.isEmpty();
+    isSourceExist ^= job.sourceQuery == null || job.sourceQuery.isEmpty();
+    if(!isSourceExist) {
+      project.logger.lifecycle("RequiredFieldsChecker ERROR: ${job.getClass().getSimpleName()} ${job.name} needs either sourceTable or sourceQuery defined");
+      foundError = true;
+    }
+
+    boolean isAvroSchemaExist = false;
+    isAvroSchemaExist ^= job.avroSchemaPath == null || job.avroSchemaPath.isEmpty();
+    isAvroSchemaExist ^= job.avroSchemaInline == null || job.avroSchemaInline.isEmpty();
+
+    if(!isAvroSchemaExist) {
+      project.logger.lifecycle("RequiredFieldsChecker ERROR: ${job.getClass().getSimpleName()} ${job.name} needs either avroSchemaPath or avroSchemaInline defined");
+      foundError = true;
+    }
+  }
+
+  @Override
+  void visitJob(HdfsToEspressoJob job) {
+    foundError |= validateTrue(job, "qps should be a positive number", job.qps > 0);
+    foundError |= validateNotEmpty(job, "sourceHdfsPath", job.sourceHdfsPath);
+    foundError |= validateNotEmpty(job, "espressoEndpoint", job.espressoEndpoint);
+    foundError |= validateNotEmpty(job, "espressoDatabaseName", job.espressoDatabaseName);
+    foundError |= validateNotEmpty(job, "espressoTableName", job.espressoTableName);
+    foundError |= validateNotEmpty(job, "espressoContentType", job.espressoContentType);
+    foundError |= validateNotEmpty(job, "espressoKey", job.espressoKey);
+    foundError |= validateNotEmpty(job, "espressoOperation", job.espressoOperation);
+    foundError |= validateNotEmpty(job, "errorHdfsPath", job.errorHdfsPath);
+  }
+
```

Comment:
Are you missing a visitJob(GobblinJob) here, and does the GobblinJob have any required properties that are specific for it?